web-server/v0.4/e2etests

```diff
@@ -61,8 +61,8 @@ else
     printf -- "... \"yarn start\" failed to start web server (%s).\n" ${res} 
 fi
 # The yarn start process leaves many processes lying around unless we
-# explicitly kill them.
-kill -s TERM -- ${yarn_start_pid}
+# explicitly kill them. Kill all processes in our process group except for the current process.
+kill -s TERM -- $(pgrep -g 0 | grep -v ^"$$"$) 2>/dev/null
```

Comment:
So the `pgrep -g 0` lists all "pids" which are in our current process group, and then ignores any lines which have `$` characters?

I tried this command sequence and it did not seem to work.

Reply:
Peter, the ^"$$"$ will exclude the pid of the process running that particular command,  i.e. when running e2etests, it will will kill everything except for e2etests itself. However, do not merge this yet because I realized this is overly aggressive, since we are killing everyone in the process group except for e2etests we will also kill the *caller* of e2etests, which is not good. I will fix this today.

Reply:
@portante here's why you were not seeing the behavior I was seeing (e2etests leaving a running web-server after exiting). On a rhel 8 image the web-server process hierarchy looks like this:

vagrant  23789 22601  0 06:48 pts/0    00:00:00           /bin/bash ./e2etests
vagrant  23813 23789  0 06:48 pts/0    00:00:00             node /usr/share/yarn/bin/yarn.js start
vagrant  23835 23813  0 06:48 pts/0    00:00:00               /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/.bin/cross-env UMI_UI=none umi dev
vagrant  23842 23835  0 06:48 pts/0    00:00:00                 /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/.bin/umi dev
vagrant  23849 23842 53 06:48 pts/0    00:00:20                   /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/umi/lib/scripts/realDev.js dev

On a bionic image used by Travis CI, the the tree looks like this:
vagrant  23080 20007  0 02:28 pts/0    00:00:00           /bin/bash ./e2etests
vagrant  23104 23080  0 02:28 pts/0    00:00:00             node /usr/share/yarn/bin/yarn.js start
vagrant  23126 23104  0 02:28 pts/0    00:00:00               /bin/sh -c cross-env UMI_UI=none umi dev
vagrant  23127 23126  0 02:28 pts/0    00:00:00                 /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/.bin/cross-env UMI_UI=none umi dev
vagrant  23134 23127  0 02:28 pts/0    00:00:00                   /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/.bin/umi dev
vagrant  23141 23134  2 02:28 pts/0    00:00:20                     /usr/bin/node /home/vagrant/pbench/web-server/v0.4/node_modules/umi/lib/scripts/realDev.js dev

Notice that the bionic image has an extra "/bin/sh" between yarn start and the web-server, so the kill of yarn will kill "node yarn" and "sh cross-env" but will not permeate past that because /bin/sh is not handling SIGTERM. There is no issue on RHEL because each node process properly cleans up a child process and there are no intervening /bin/sh processes to cause problems.

This subtle difference is because bionic links /bin/sh to dash whereas RHEL links it to bash, and these two shells differ in show they handle the node shebang (#!/usr/bin/env node).

OK so is this something worth fixing, if it only manifests in the CI?

Reply:
Seems like a simple fix to update the `.travisci.yml` file ensure the `/bin/sh` is a symlink to `bash`.  Can you consider what that would take?

Reply:
OK I will try that out @portante